### PR TITLE
[Feature] Make DQN compatible with nn.Module

### LINKED ISSUE #607: [Feature Request] Make DQN compatible with nn.Module (not wrapped in a QValue specialized network)
## Motivation

For DQNLoss to work out of the box, it'd be nice to support nn.Module (and not necessarily specialized TensorDictModule instances).

It should be fairly easy: we can already easily wrap a nn.Module in a TensorDictModule assuming that the nn.Module has only one output (see [here](https://github.com/pytorch/rl/blob/487838844b1cca2c4c200cf10b0e10a5d322de31/torchrl/collectors/collectors.py#L165-L204)). In this example, we check if the policy (in our case: the value network) is some sort of `TensorDictModule`. If not, we wrap it in that.

We could do the same with DQN: if the value network is an nn.Module that just returns a single value tensor, but does not expect a `TensorDict` as input, we should wrap it in a [`QValueActor`](https://github.com/pytorch/rl/blob/487838844b1cca2c4c200cf10b0e10a5d322de31/torchrl/modules/tensordict_module/actors.py#L409-L446) and the trick is done.

### PR DESCRIPTION
**Pull request by svarolgunes**
## Description

Made DQNLoss and DistribtutionalDQNLoss compatible with value networks that are regular nn.Module. A custom helper function is written that checks if an nn.Module has certain properties (namely in_keys and out_keys that match a certain signature). If there is a signature mismatch, it throws an error. If there are no in_keys and out_keys attributes, then the nn.Module is wrapped in the given wrapper_type with the (optional) desired in_keys and out_keys list.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
`close #607 `

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/objectives/dqn.py [hunk 1]
```diff
@@ -30,7 +34,7 @@ class DQNLoss(LossModule):
 
     def __init__(
         self,
-        value_network: QValueActor,
+        value_network: Union[QValueActor, nn.Module],
```
Comment (vmoens) [inline torchrl/objectives/dqn.py:37]: we should update the docstrings

**Review by vmoens — COMMENTED** [review]
LGTM, can you have a look at where codecov detected that tests were missing and code up a fix if appropriate?
To catch errors, don't hesitate to group the tests in a single function.

File: torchrl/modules/tensordict_module/common.py [hunk 2]
```diff
@@ -603,3 +605,83 @@ def __getattr__(self, name: str) -> Any:
 
     def forward(self, *args, **kwargs):
         return self.td_module.forward(*args, **kwargs)
+
+
+def is_tensordict_compatible(module: Union[TensorDictModule, nn.Module]):
```
[the remark this answers is not in the document]
Comment (svarolgunes) [reply torchrl/modules/tensordict_module/common.py:610]: I've tried to address adding util functions to the doc but I'm not sure whether I've put them in the right place or not.

File: torchrl/objectives/dqn.py (hunk 1 above)
Comment (svarolgunes) [reply torchrl/objectives/dqn.py:37]: Docstrings are updated accordingly.

File: torchrl/modules/tensordict_module/common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (svarolgunes) [reply torchrl/modules/tensordict_module/common.py:610]: Tests are added.

**Review by vmoens — APPROVED** [review]
LGTM thanks for this!

**Pull request merged**